src/extension.ts

```diff
@@ -51,6 +54,22 @@ async function init(context: vscode.ExtensionContext, git: GitApiImpl, credentia
 		}
 	});
 
+	context.subscriptions.push(git.onDidPublish(async e => {
+		// Only notify on branch publish events
+		if (!e.branch || PersistentState.fetch(PROMPTS_SCOPE, PROMPT_TO_CREATE_PR_ON_PUBLISH_KEY) === false) {
```

Comment:
I think we should also avoid showing this notification if you're already in the create view - if you don't have an upstream for you current branch, we show a button that will run the publish action from there.

You could move this to `reviewManager` to have access to the `CreatePullRequestHelper` there, and also direct access to the create method instead of executing the command. `CreatePullRequestHelper` doesn't expose anything right now that indicates if create is in progress, so that would need to be added

Reply:
👌👌